src/main/kotlin/com/fatimacarneiro/estudokotlin/phonenumber/Service.kt

```diff
@@ -0,0 +1,57 @@
+package com.fatimacarneiro.estudokotlin.phonenumber
+
+import com.fatimacarneiro.estudokotlin.DDDException
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Transactional
+
+@Service
+class Service(
+        val dao: Dao,
+        val conversor: Conversor
+) {
+
+    @Transactional
+    fun save(form: PhoneNumberForm) {
+        val entity: PhoneNumber = conversor.toConvertFormInEntity(form)
+        if (!validPrefix(form.ddd)) throw DDDException("Invalid prefix")
+
+        dao.save(entity)
+    }
+
+//    fun save(form: PhoneNumberForm) =
+//            with(conversor.toConvertFormInEntity(form)) {
+//                this.takeIf {
+//                    validPrefix(it.ddd)
+//                }?.let {
+//                    dao.save(it)
+//                } ?: throw DDDException("Invalid prefix")
+//            }
+
+//    fun save(form: PhoneNumberForm) =
+//            with(conversor.toConvertFormInEntity(form)) {
+//                if (!validPrefix(this.ddd))
+//                    throw DDDException("Invalid prefix")
+//                dao.save(this)
+//            }
+
+
+    fun validPrefix(dddForm: Int): Boolean {
+        return DDD.values().asList().stream().anyMatch { ddd ->
+            ddd.prefix == dddForm
+        }
+    }
+
+    fun getAll(): MutableList<View> {
+        val views = mutableListOf<View>()
+        dao.findAll().forEach {
```

Comment:
if you change `for each` for `map`, the result is a list and you dont't need add manually